package/scripts/prmon_plot.py

```diff
@@ -257,24 +306,29 @@ def main():
 
     # Here comes the figure and data extraction
     fig, ax1 = plt.subplots()
-    xdata = np.array(data[xlabel]) * xmultiplier
+
+    xdata = []
     ydlist = []
-    for carg in ylist:
-        if args.diff:
-            num = np.array(data[carg].diff()) * ymultiplier
-            denom = np.array(data[xlabel].diff()) * xmultiplier
-            ratio = np.where(denom != 0, num / denom, np.nan)
-            ydlist.append(ratio)
-        else:
-            ydlist.append(np.array(data[carg]) * ymultiplier)
+
+    for i in range(len(data)):
+        xdata.append(np.array(data[i][xlabel] * xmultiplier))
+        ydlist.append(make_list(ylist, data[i], args, xmultiplier, ymultiplier, xlabel))
+
+    # Plot the graphs
+    line_styles = list(mpl.lines.lineStyles.keys())
+
     if args.stacked:
-        ydata = np.vstack(ydlist)
-        plt.stackplot(
-            xdata, ydata, lw=2, labels=[LEGENDNAMES[val] for val in ylist], alpha=0.6
-        )
+        if len(inputs) == 1:
+            for i in range(len(xdata)):
+                draw_stacked_graph(xdata[i], ydlist[i], ylist)
+        else:
+            print("ERROR:: Stacked graphs are not supported for more than one data set")
+            sys.exit(-1)
     else:
-        for cidx, cdata in enumerate(ydlist):
-            plt.plot(xdata, cdata, lw=2, label=LEGENDNAMES[ylist[cidx]])
+        for i in range(len(xdata)):
+            draw_line_graph(xdata[i], ydlist[i], ylist, line_styles[i%7], inputs, i)
```

Comment:
This line fails a [flake8 test](https://www.flake8rules.com/rules/E228.html). It should be:

```py
draw_line_graph(xdata[i], ydlist[i], ylist, line_styles[i % 7], inputs, i)
```

We should fix that too.

Reply:
Fixed!